src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -409,4 +408,37 @@ public void resetOrderView() {
         updateFilteredOrderList(PREDICATE_SHOW_ALL_ORDERS);
     }
 
+    //=========== AddressBook & OrderBook Relation Check =======================================================
+
+    /**
+     * Checks if any order tagged to persons that don't exist.
+     */
+    public void checkClientAndOrderRelation() throws DataConversionException {
+        ObservableList<Order> orders = this.orderBook.getOrderList();
+        for (Order eachOrder : orders) {
+            String nameOfPerson = eachOrder.getCustomer().getName();
+            if (!this.addressBook.hasPersonWithName(nameOfPerson)) {
+                throw new DataConversionException(
+                        new IllegalValueException("Given Client Name does not exist in Address Book"));
```

Comment:
I think it's better to use 'customer name` here to make the error message clearer.